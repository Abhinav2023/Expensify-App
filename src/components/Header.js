import React from 'react';
import {NavLink } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';


export const Header=({startLogout})=>(
    <div>
        <h1>Expensify App</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home Page</NavLink>
        <NavLink to="/create" activeClassName="is-active">Go To Create Page</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help Page</NavLink>
        <button onClick={startLogout}>Logout</button>
  </div>
);

const mapDispatchToProps=(dispatch)=>({
  startLogout: ()=>dispatch(startLogout())
})

export default connect(undefined,mapDispatchToProps)(Header)