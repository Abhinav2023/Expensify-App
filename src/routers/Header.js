import React from 'react';
import { BrowserRouter, Route,Switch,Link,NavLink } from 'react-router-dom';
const Header=()=>(
    <div>
        <h1>Expensify App</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home Page</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Go To Edit Page</NavLink>
        <NavLink to="/create" activeClassName="is-active">Go To Create Page</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help Page</NavLink>
  </div>
);
export default Header;