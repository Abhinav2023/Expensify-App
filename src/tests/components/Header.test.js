import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Header from '../../components/Header';
import { exportAllDeclaration } from '@babel/types';


test("should render Header correctly",()=>{
    const wrapper=shallow(<Header />);
    expect(wrapper.find('h1').length.text()).toBe("Expensify App")
    // const renderer =new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
})
