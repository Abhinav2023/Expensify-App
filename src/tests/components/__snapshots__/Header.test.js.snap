// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div>
  <h1>
    Expensify App
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    Home Page
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Go To Create Page
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/help"
  >
    Help Page
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
